### PR TITLE
Update button

### PR DESCRIPTION
**Pull request by igorkorenfeld**
Option 2 in #169 , adding in an apply filters button 

- Adds apply button on change of selected filters
- Adjusts styling of the filters section to make it more distinct from the main section
- Limits height of the filters sidebar so the user can see the content didn't refresh 

Also tries to address some of the a11y @iamjolly flagged for us:
- #148  - Added an aria-live section for the filters 
- #149 - Changed filter badge image to be a button with "remove {filter name}" as the alt text

image

### CODE REVIEW

**Comment by peterrowland** [comment]
TODOs:
NaN re-appearing
why does the page reload when clearing all agency/section filters?
Should it show apply filters when date filters are removed?

**Comment by igorkorenfeld** [comment]
From  the conversation with @peterrowland 
Outline for tests

### 1
- select agency checkbox
- select section checkbox
- expect 

### 2
- add date range 
- expect applyBtn

### 3   
- select agnecy checkbox
- deselect same agency checkbox
- expect applyBtn.length === 0
### 4
- After load with filters
- expect applyBtn.length === 0
- No apply 


Also need to check on behavior for when "Clear all" buttons is pressed with respect to apply filters button

File: app/assets/stylesheets/application.scss [hunk 1]
```diff
@@ -126,20 +127,38 @@ $font-stack: Source Sans Pro Web,Helvetica Neue,Helvetica,Roboto,Arial,sans-seri
...
 .active-filter {
   display: inline-block;
   border-radius: 0;
-  padding: 5px;
-  padding-right: 8px;
-  padding-left: 8px;
-  margin-bottom: 17px;
+  padding: 0.25em;
+  padding-right: 0.5em;
+  padding-left: 0.5em;
```
Comment (lauraGgit) [inline app/assets/stylesheets/application.scss:153]: you can shorten this by changing it to `padding: 0.25em 0.5em`

File: app/javascript/packs/search.js [hunk 2]
```diff
@@ -27,32 +43,84 @@ $( function () {
...
+    let currentStartDate = parseInt($("#starting_year").val());
+    let currentEndDate = parseInt($("#ending_year").val());
+    if(publicationDateValidation()) {
+      if (isNaN(initialStartDate) && isNaN(currentStartDate) && isNaN(initialEndDate) && isNaN(currentEndDate)) {
+        dateChangeFlag = false;
+      }
+      else if (currentStartDate === initialStartDate && currentEndDate === initialEndDate) {
+        dateChangeFlag = false;
+      }
+      else {
```
Comment (lauraGgit) [inline app/javascript/packs/search.js:95]: to DRY this up you could initialize the value as false and then only make a single conditional that sets the value to `true`

File: app/javascript/packs/search.js [hunk 3]
```diff
@@ -27,32 +43,84 @@ $( function () {
...
+  $("#starting_year").on("change", detectDateChange);
+  $("#ending_year").on("change", detectDateChange);
+
+  // Detects if the change to publication date range is different from load
+  // and displays the apply button if so
+  function detectDateChange() {
+    let currentStartDate = parseInt($("#starting_year").val());
+    let currentEndDate = parseInt($("#ending_year").val());
+    if(publicationDateValidation()) {
+      if (isNaN(initialStartDate) && isNaN(currentStartDate) && isNaN(initialEndDate) && isNaN(currentEndDate)) {
```
Comment (lauraGgit) [inline app/javascript/packs/search.js:89]: it seems like you might effectively be doing this date validation twice on the current dates. you might want to consider writing it to only need to check once.

File: app/javascript/packs/search.js [hunk 4]
```diff
@@ -141,17 +209,26 @@ function includeFilteredCheckedAgenciesInSearch(agencyList) {
...
 }
 
+// Validates publication date range and creates badge.
+// returns True if successful, false otherswise
 function publicationDateValidation(){
   let startYear = parseInt($("#starting_year").val());
   let endYear = parseInt($("#ending_year").val());
+  if (isNaN(startYear) && isNaN(endYear)) {
+    clearDatesFilter();
+    return true;
```
Comment (lauraGgit) [inline app/javascript/packs/search.js:219]: given that there are 4 cases here, I would initialize a variable as either `true/false` and then update the cases to return that variable as that could dry this up a bit.

**Review by lauraGgit — COMMENTED** [review]
haven't had a chance to review locally, but here are some initiall comments

**Review by ondrae — CHANGES_REQUESTED** [review]
If I type something into the agency filter box and hit the enter button, all the buttons and links that would submit get disabled.
Screen Shot 2021-02-26 at 2 30 02 PM

Update: Oooooh it's an existing bug. It happens on production too.

File: app/javascript/packs/search.js [hunk 5]
```diff
@@ -1,24 +1,40 @@
...
     }
   });
 
   // Get :checked on load and create badges
-  $(".sidebar :checked").each(function(){
+  const checked = $(".sidebar :checked");
+  if (checked.length > 0) {
+    if($("#active-filters").is(":hidden")) {
+      $("#active-filters").show();
+    }
```
Comment (ondrae) [inline app/javascript/packs/search.js:35]:
```suggestion
    showActiveFilters();
```

This method was added below, use it here.

**Review by ondrae — CHANGES_REQUESTED** [review]
This `showActiveFilters` method needs an opposite too. I ended up with a blank `#active-filters` area by adding some filters then removing them, before submitting.

Screen Shot 2021-02-26 at 3 02 51 PM

A test for this can go in the spec/system/search_spec.rb file, adding some filters, removing them then asserting `expect(page).to have_selector("#active-filters", visible: false)`

**Comment by igorkorenfeld** [comment]
@ondrae should #173 be merged before this one, this one first, or either way?

**Review by ondrae — APPROVED** [review]
Great work @igorkorenfeld

**Pull request merged**